From 3e98aaf9e5c6b2206edce3309beb1adeb2b61b60 Mon Sep 17 00:00:00 2001
From: Luo Jie <quic_luoj@quicinc.com>
Date: Thu, 23 Nov 2023 15:41:20 +0800
Subject: [PATCH 15/17] arm64: dts: qcom: Add IPQ9574 MDIO device node

The MDIO bus master block is used to accessing the MDIO slave
device (such as PHY device), the dedicated MDIO PINs needs to
be configured.

Change-Id: Ia64083529e693256dbd8f8af4071c02afdded8f9
Signed-off-by: Luo Jie <quic_luoj@quicinc.com>
---
 arch/arm64/boot/dts/qcom/ipq9574.dtsi | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

--- a/arch/arm64/boot/dts/qcom/ipq9574.dtsi
+++ b/arch/arm64/boot/dts/qcom/ipq9574.dtsi
@@ -251,6 +251,8 @@
 		mdio: mdio@90000 {
 			compatible =  "qcom,ipq9574-mdio", "qcom,ipq4019-mdio";
 			reg = <0x00090000 0x64>;
+			pinctrl-0 = <&mdio_pins>;
+			pinctrl-names = "default";
 			#address-cells = <1>;
 			#size-cells = <0>;
 			clocks = <&gcc GCC_MDIO_AHB_CLK>;
@@ -322,6 +324,22 @@
 			interrupt-controller;
 			#interrupt-cells = <2>;
 
+			mdio_pins: mdio-pins {
+				mdc-state {
+					pins = "gpio38";
+					function = "mdc";
+					drive-strength = <8>;
+					bias-disable;
+				};
+
+				mdio-state {
+					pins = "gpio39";
+					function = "mdio";
+					drive-strength = <8>;
+					bias-pull-up;
+				};
+			};
+
 			uart2_pins: uart2-state {
 				pins = "gpio34", "gpio35";
 				function = "blsp2_uart";
